### ISSUE TITLE
What's the proper way to access database/api?

### ISSUE THREAD

**Issue by UROjQ6r80p** [marko-js/run#42]
Let's say I want to fetch products from the database.

that's the old way:

```
import express from "express";
import markoPlugin from "@marko/express";
import template from "./template.marko";

const app = express();
app.use(markoPlugin()); // Enables `res.marko(template, input)`

app.get("/", function (req, res) {
  const products = req.db.find(...)

  res.marko(template, {
    products
  });
});

app.listen(8080);
```

What's the proper way to do the same in marko/run?

**Comment by vwong** [comment]
I don't think this is the right forum to ask support questions like these; try https://discord.com/invite/marko next time.

But in short, I think the best way to do this is to add a `+middleware.js` early on, that appends your database connector to `context` aka `$global`, using something like `context.dbClient = dbClient`. This way, you only ever need to create `+page.marko` with `<await($global.dbClient.find(...))>` and don't have to litter your code with `+handler.js` everywhere.
